Description: Fix wordsize detection for x32 architecture
 This is intended to make "MARISA_WORD_SIZE 32" being set when
 building on x32, and with that prevent a FTBFS due to a failed
 base-test. Idea from https://wiki.debian.org/X32Port
Author: Gunnar Hjalmarsson <gunnarhj@ubuntu.com>
Forwarded: no
Last-Update: 2021-02-09

--- a/include/marisa/base.h
+++ b/include/marisa/base.h
@@ -28,7 +28,8 @@
 typedef uint64_t marisa_uint64;
 #endif  // _MSC_VER
 
-#if defined(_WIN64) || defined(__amd64__) || defined(__x86_64__) || \
+#if defined(_WIN64) || ( defined(__amd64__) && !defined(__ILP32__) ) || \
+    ( defined(__x86_64__) && !defined(__ILP32__) ) || \
     defined(__ia64__) || defined(__ppc64__) || defined(__powerpc64__) || \
     ( defined(__sparc__) && defined(__arch64__) ) || \
     ( defined(__riscv) && (__riscv_xlen == 64) ) || \
